crates/codegen/parser/runtime/src/support/scanner_macros.rs

```diff
@@ -66,6 +66,26 @@ macro_rules! scan_choice {
     };
 }
 
+#[allow(unused_macros)]
+macro_rules! scan_keyword_choice {
+    ($stream:ident, $ident:ident, $($scanner:expr),*) => {
+        loop {
+            let save = $stream.position();
+            $(
+                {
+                    if let result @ (KeywordScan::Present(..) | KeywordScan::Reserved(..)) = ($scanner) {
+                        if $ident.len() == $stream.position().utf8 - save.utf8 {
```

Comment:
Not perfect, I admit; this should probably be handled with an automaton for the keywords and executed/matched on the string till it's exhausted but let's punt on this for now